traffic_ops/traffic_ops_golang/deliveryservice/keys.go

```diff
@@ -135,6 +144,48 @@ func GetSSLKeysByHostName(w http.ResponseWriter, r *http.Request) {
 	}
 
 	hostName := inf.Params["hostname"]
+	xmlID, ok, err := getXmlIdFromHostname(inf, hostName)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	if !ok {
+		api.WriteRespAlert(w, r, tc.InfoLevel, "- error getting cdn or delivery service for hostname "+hostName)
+		return
+	}
+
+	getSSLKeysByXMLIDHelper(xmlID, inf, w, r)
+}
+
+// GetSSLKeysByHostName fetches the ssl keys for a deliveryservice specified by the fully qualified hostname. V14 includes expiration date.
+func GetSSLKeysByHostNameV14(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"hostname"}, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	if inf.Config.RiakEnabled == false {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusServiceUnavailable, errors.New("the Riak service is unavailable"), errors.New("getting SSL keys from Riak by host name: Riak is not configured"))
```

Comment:
It's been decided since this PR was opened that the configuration of a sub-service of TO should not be exposed in any way - meaning that in this case it should log the error but only tell the client that a generic internal server error occurred.

Reply:
done